packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -131,7 +137,20 @@ export function createReactExtension<
         return (
           <Suspense fallback={<Progress />}>
             <PluginErrorBoundary app={app} plugin={plugin}>
-              <Component {...props} />
+              <AnalyticsContext
+                attributes={{
+                  pluginId: plugin.getId(),
+                  ...(options.name ? { extension: options.name } : {}),
+                  ...(data['core.mountpoint']
+                    ? {
+                        routeRef: (data['core.mountpoint'] as { id?: string })
+                          .id,
+                      }
+                    : {}),
+                }}
```

Comment:
Suggest creating this attributes object outside of the jsx, it becomes so large and hard to read in here.

Also a nicety with js: object spread is really resilient against "improper" things to spread, which it just ignores silently. You can therefore for example do

```ts
...(options.name && { extension: options.name })
// vs
...(options.name ? { extension: options.name } : {});
```

but that's highly a matter of taste :)